docs/QSG/qemu-vms.md

```diff
@@ -0,0 +1,285 @@
+        if do_brctl show | grep "^$1" > /dev/null 2> /dev/null; then
+        return 1
+        else
+        return 0
+        fi
+    }
+    
+    create_bridge() {
+        do_brctl addbr "$1"
+        do_brctl stp "$1" off
+        do_brctl setfd "$1" 0
+        do_ifconfig "$1" "$GATEWAY" netmask "$NETMASK" up
+    }
+    
+    enable_ip_forward() {
+        echo 1 | do_dd of=/proc/sys/net/ipv4/ip_forward > /dev/null
+    }
+    
+    add_filter_rules() {
+    do_iptables_restore <<EOF
+    # Generated by iptables-save v1.3.6 on Fri Aug 24 15:20:25 2007
+    *nat
+    :PREROUTING ACCEPT [61:9671]
+    :POSTROUTING ACCEPT [121:7499]
+    :OUTPUT ACCEPT [132:8691]
+    -A POSTROUTING -s $NETWORK/$NETMASK -j MASQUERADE 
+    COMMIT
+    # Completed on Fri Aug 24 15:20:25 2007
+    # Generated by iptables-save v1.3.6 on Fri Aug 24 15:20:25 2007
+    *filter
+    :INPUT ACCEPT [1453:976046]
+    :FORWARD ACCEPT [0:0]
+    :OUTPUT ACCEPT [1605:194911]
+    -A INPUT -i $BRIDGE -p tcp -m tcp --dport 67 -j ACCEPT 
+    -A INPUT -i $BRIDGE -p udp -m udp --dport 67 -j ACCEPT 
+    -A INPUT -i $BRIDGE -p tcp -m tcp --dport 53 -j ACCEPT 
+    -A INPUT -i $BRIDGE -p udp -m udp --dport 53 -j ACCEPT 
+    -A FORWARD -i $1 -o $1 -j ACCEPT 
+    -A FORWARD -s $NETWORK/$NETMASK -i $BRIDGE -j ACCEPT 
+    -A FORWARD -d $NETWORK/$NETMASK -o $BRIDGE -m state --state RELATED,ESTABLISHED -j ACCEPT 
+    -A FORWARD -o $BRIDGE -j REJECT --reject-with icmp-port-unreachable 
+    -A FORWARD -i $BRIDGE -j REJECT --reject-with icmp-port-unreachable 
+    COMMIT
+    # Completed on Fri Aug 24 15:20:25 2007
+    EOF
+    }
+    
+    start_dnsmasq() {
+        do_dnsmasq \
+        --strict-order \
+        --except-interface=lo \
+        --interface=$BRIDGE \
+        --listen-address=$GATEWAY \
+        --bind-interfaces \
+        --dhcp-range=$DHCPRANGE \
+        --conf-file="" \
+        --pid-file=/var/run/qemu-dnsmasq-$BRIDGE.pid \
+        --dhcp-leasefile=/var/run/qemu-dnsmasq-$BRIDGE.leases \
+        --dhcp-no-override \
+        ${TFTPROOT:+"--enable-tftp"} \
+        ${TFTPROOT:+"--tftp-root=$TFTPROOT"} \
+        ${BOOTP:+"--dhcp-boot=$BOOTP"}
+    }
+    
+    setup_bridge_nat() {
+        if check_bridge "$1" ; then
+        create_bridge "$1"
+        enable_ip_forward
+        add_filter_rules "$1"
+        start_dnsmasq "$1"
+        fi
+    }
+    
+    setup_bridge_vlan() {
+        if check_bridge "$1" ; then
+        create_bridge "$1"
+        start_dnsmasq "$1"
+        fi
+    }
+    
+    setup_bridge_nat "$BRIDGE"
+    
+    if test "$1" ; then
+        do_ifconfig "$1" 0.0.0.0 up
+        do_brctl addif "$BRIDGE" "$1"
+    fi
+```
+Then add a sudo rule for qemu-system-x86<sub>64</sub> in *etc/sudoers.d*.
+```
+    <username> blackbox=(root) NOPASSWD: /usr/bin/qemu-system-x86_64
+```
+3.  Install guest operating systems.
+
+Run these two comands. Select "boot from cdrom". Follow the installation guide to install the OSes.
```

Comment:
@paul356, I was able to run the CI and the following spelling error was detected.
Please, could you fix it.
Thanks in advance.
```suggestion
Run these two commands. Select "boot from cdrom". Follow the installation guide to install the OSes.
```

Reply:
Sorry for late response. I was on vacation for previous days. I will fix it today.